graphql_compiler/compiler/ir_lowering_sql/__init__.py

```diff
@@ -32,4 +40,187 @@ def lower_ir(ir_blocks, query_metadata_table, type_equivalence_hints=None):
+    return query_path_to_location_info
+
+
+def _location_infos_equal(left, right):
+    """Return True if LocationInfo objects are equivalent for the SQL backend, False otherwise.
+
+    LocationInfo objects are considered equal for the SQL backend iff the optional scopes depth,
+    recursive scopes depth, types and parent query paths are equal.
+
+    Args:
+        left: LocationInfo, left location info object to compare.
+        right: LocationInfo, right location info object to compare.
+
+    Returns:
+        bool, True if LocationInfo objects equivalent, False otherwise.
+    """
+    if not isinstance(left, LocationInfo) or not isinstance(right, LocationInfo):
+        raise AssertionError(
+            u'Unsupported LocationInfo comparison between types {} and {} '
+            u'with values {}, {}'.format(type(left), type(right), left, right))
+    optional_scopes_depth_equal = (left.optional_scopes_depth == right.optional_scopes_depth)
+
+    parent_query_paths_equal = (
+        (left.parent_location is None and right.parent_location is None) or
+        (left.parent_location.query_path == right.parent_location.query_path))
+
+    recursive_scopes_depths_equal = (left.recursive_scopes_depth == right.recursive_scopes_depth)
+
+    types_equal = left.type == right.type
+
+    return all([
+        optional_scopes_depth_equal,
+        parent_query_paths_equal,
+        recursive_scopes_depths_equal,
+        types_equal,
+    ])
+
+
+def _map_query_path_to_outputs(construct_result, query_path_to_location_info):
+    """Assign the output fields of a ConstructResult block to their respective query_path."""
+    query_path_to_output_fields = {}
+    for output_name, field in six.iteritems(construct_result.fields):
+        field_name = field.location.field
+        output_query_path = field.location.query_path
+        output_field_info = constants.SqlOutput(
+            field_name=field_name,
+            output_name=output_name,
+            graphql_type=query_path_to_location_info[output_query_path].type)
+        output_field_mapping = query_path_to_output_fields.setdefault(output_query_path, [])
+        output_field_mapping.append(output_field_info)
+    return query_path_to_output_fields
+
+
+def _map_block_index_to_location(ir_blocks):
+    """Associate each IR block with its corresponding location, by index."""
+    block_index_to_location = {}
+    # MarkLocation blocks occur after the blocks related to that location.
+    # The core approach here is to buffer blocks until their MarkLocation is encountered
+    # after which all buffered blocks can be associated with the encountered MarkLocation.location.
+    current_block_ixs = []
+    global_operations_started = False
+    for num, ir_block in enumerate(ir_blocks):
+        if global_operations_started:
+            continue
+        if isinstance(ir_block, blocks.GlobalOperationsStart):
+            global_operations_started = True
```

Comment:
Why set the flag here, and then `continue` on the next iteration? This approach will add the `GlobalOperationsStart` block to `current_block_ixs` before breaking out of the loop, which doesn't seem like a good effect. Why not just a `break` at the bottom of the conditional that starts on line 214?

Reply:
Good point, changed to break.